components/buildless-serverless/internal/controller/cache/cache.go

```diff
@@ -0,0 +1,63 @@
+package cache
+
+import (
+	"sync"
+	"time"
+)
+
+var (
+	_ InMemoryCache = (*repoLastCommitCache)(nil)
+)
+
+type InMemoryCache interface {
+	Set(any, string)
+	Get(any) *string
+	Delete(any)
+}
+
+// repoLastCommitCache provides an in-memory processor to store buildless serverless key values pairs with timestamp. By using sync.Map for caching,
+// concurrent operations to the processor from diverse reconciliations are considered safe.
+//
+// Inside the processor is stored last commit with which was git function created by repository url and reference as the key.
+type repoLastCommitCache struct {
+	processor sync.Map
+	timeout   time.Duration
+}
+
+type storageObject struct {
+	timestamp time.Time
+	value     string
+}
+
+// NewInMemoryCache returns a new instance of repoLastCommitCache.
+func NewInMemoryCache(timeout time.Duration) *repoLastCommitCache {
```

Comment:
I'd suggest renaming this method to `NewRepoLastCommitCache`

Reply:
Done as suggested